pcdsdevices/beam_stats.py

```diff
@@ -1,18 +1,17 @@
 import logging
 
 from ophyd.device import Component as Cpt
-from ophyd.device import Device
 from ophyd.signal import AttributeSignal, EpicsSignal, EpicsSignalRO
 
+from .device import GroupDevice
 from .interface import BaseInterface
 from .pv_positioner import PVPositionerDone
 from .signal import AvgSignal
 
-
 logger = logging.getLogger(__name__)
 
 
-class BeamStats(BaseInterface, Device):
```

Comment:
I went back and forth on this one and a few others multiple times. I ended up making it a GroupDevice because of the differing IOC sources, but in practice nothing is done to an EpicsSignal on stage... so the decision is pretty moot. This doesn't even affect connection checking. It doesn't do anything at all I'm pretty sure. Maybe if it doesn't do anything at all it's better to reduce complexity and I should change singnal-only devices back.

Reply:
I reflected on this a bit more since looking at the code - and I agree that it shouldn't really change anything either way at the device level.

But let's go one level up: it _may_ influence what happens in typhos, when that future GroupDevice template comes about. In which case, you could either design that template around this case being OK, or just not allowing it to start with. Hard to say until you get there, but the latter option may be slightly easier.

Reply:
Oh that's a really good point. With that logic, these should definitely not be `GroupDevice`s, the template will assume that we have 2+ subdevices and deprioritize the top-level signals.

Reply:
I've identified 4 devices to change back:
- BeamStats (all sigs)
- LCLS (all sigs)
- MPODApalisModule (all sigs)
- IonPumpWithController (mostly sigs, one subdevice)